src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs

```diff
@@ -384,6 +384,8 @@ public sealed class SourcesCommandConfiguration
         public string Username { get; set; }
         public string Password { get; set; }
         public int Priority { get; set; }
+        public string Certificate { get; set; }
+        public string CertificatePassword { get; set; }

```

Comment:
Shouldn't the certificate be validated with the remote server or am I missing something? 

So what you are stating is that it is likely in a single call for a source it will use both the certificate AND the username/password?